site/profile/manifests/core/rucio.pp

```diff
@@ -1,75 +1,70 @@
-# @summary
-#   Install required rucio packages
-#
-class profile::core::rucio () {
-  include profile::core::letsencrypt
-
-  #  Host FQDN
-  $fqdn = $facts[fqdn]
-
-  #  Define XRootD Path
-  $xrootd_path = '/opt/xrootd'
-
-  #  Define Yum Packages
-  $yum_packages = [
-    'gcc-c++',
-    'cmake3',
-    'krb5-devel',
-    'libuuid-devel',
-    'libxml2-devel',
-    'openssl-devel',
-    'systemd-devel',
-    'zlib-devel',
-    'devtoolset-7',
-    'xrootd',
-    'voms',
-  ]
+class profile::core::rucio {
+  # Ensure the GPG key is imported
+  exec { 'import_xrootd_gpg_key':
+    command => '/usr/bin/rpm --import https://xrootd.web.cern.ch/repo/RPM-GPG-KEY.txt',
+  }
 
-  #  Define PIP Packages
-  $pip_packages = [
-    'wheel',
-    'cryptography',
-    'rucio',
-  ]
+  # Fetch the xrootd.repo file
+  exec { 'fetch_xrootd_repo':
```

Comment:
A good rule of thumb is that if `exec` is being used, it probably isn't the correct approach.  The `yumrepo` type should be configure a yum repo.